### PR TITLE
Log click events when visiting shortlinks.

### PR DESCRIPTION
**Pull request by DFurnes**
### What's this PR do?

This pull request updates our "visit link" route to write click events to our storage bucket, using the [same schema](https://docs.google.com/document/d/1CXHiGctESpc-tPYq6wYCLGKCSyzRm_izSaLy6GH-Oik/edit#heading=h.dp032g1c0uw1) as rows in our existing PostgreSQL clicks database:

Screen Shot 2020-05-14 at 11 51 18 AM

Here's an example of one of those click log files:

```
{
  "click_id": "w48OKyE-ea8658bc-c7c5-467f-9203-c68c5d92a850",
  "shortened": "w48OKyE",
  "target_url": "https://www.dfurnes.com",
  "click_time": "2020-05-14T15:50:57.244Z",
  "user_agent": "curl/7.69.1"
}
```

### How should this be reviewed?

🍃 Cleanup: I extracted the Dynamoose model config into `config/database.js` in 5a29624.

💾 I installed & configured [Flydrive](https://github.com/Slynova-Org/flydrive), a [Flysystem](https://flysystem.thephpleague.com/v1/docs/)-inspired storage driver for Node.js. This allows us to write these files to disk on local development, and to an S3 bucket in production.

📝 Finally, I added the code to write these click logs in ebc63d1.

### Any background context you want to provide?

Yes, I'm shamelessly trying to re-build Laravel (or at least Lumen) in Node.js. 😇

### Relevant tickets

References [Pivotal #172799447](https://www.pivotaltracker.com/story/show/172799447).

### Checklist

- [ ] This PR has been added to the relevant Pivotal card.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: config/database.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    update: false,
+
+    // We use 'ON_DEMAND' capacity by default since load is so variable.
+    throughput: 'ON_DEMAND',
+
+    // Prefix our DynamoDB tables with the application name,
+    // e.g. 'bertly-links' for the 'links' table.
+    prefix: `${process.env.APP_NAME || 'bertly'}-`,
+  },
+};
```
Comment (sheyd) [inline config/database.js:28]: Love the refactor!

**Comment by DFurnes** [comment]
> Only thing I couldn't track down (a n3w day, a n3w n00b question), where do we specify if `local` vs `s3` storage driver is used in dev vs. prod?

Oops, forgot to expose an environment variable for that! It's the `filesystems.default` config var.

**Comment by DFurnes** [comment]
Added the ability to swap that with the `STORAGE_DRIVER` environment variable in 2c73356.

**Comment by weerd** [comment]
```
{
  "click_id": "w48OKyE-ea8658bc-c7c5-467f-9203-c68c5d92a850",
  "shortened": "w48OKyE",
  "target_url": "https://www.dfurnes.com",
  "click_time": "2020-05-14T15:50:57.244Z",
  "user_agent": "curl/7.69.1"
}
```

trying to pad your dfurnes.com google analytics eh? 🤔

File: src/Models/Link.js [hunk 2]
```diff
@@ -30,22 +30,9 @@ const schema = new Schema(
...
-  // TODO: This seems to be bugged <https://git.io/Jf401>. We should see if we
-  // can figure out what's going wrong here so we can run "update" migrations.
-  update: false,
-
-  // We use 'ON_DEMAND' capacity for this table since load is so variable.
-  throughput: 'ON_DEMAND',
-
-  // Prefix DynamoDB tables with the application name.
-  prefix: `${config('app.name')}-`,
+  ...config('database.defaults'),
```
Comment (weerd) [inline src/Models/Link.js:35]: 👏

**Comment by DFurnes** [comment]
> trying to pad your dfurnes.com google analytics eh? 🤔

Gotta get that SEO!!

**Pull request merged**